h2o-core/src/main/java/water/JavaVersionSupport.java

```diff
@@ -0,0 +1,45 @@
+package water;
+
+
+import static water.util.JavaVersionUtils.JAVA_VERSION;
+
+public class JavaVersionSupport {
+    // Notes: 
+    // Inclusive interval for supported Java versions.
+    // Make sure that the following is logically consistent with whitelist in R code - see function .h2o.check_java_version in connection.R.
+    // Upgrade of the javassist library should be considered when adding support for a new java version.
+    public static final int MIN_SUPPORTED_JAVA_VERSION = 8;
+    public static final int MAX_SUPPORTED_JAVA_VERSION = 14;
+
+    /**
+     * Checks for the version of Java this instance of H2O was ran with and compares it with supported versions.
+     *
+     * @return True if the instance of H2O is running on supported JVM, otherwise false.
+     */
+    public static boolean runningOnSupportedVersion() {
+        return JAVA_VERSION.isKnown() && !isUserEnabledJavaVersion() && isSupportedVersion();
```

Comment:
This condition is wrong. Thanks for revealing it @satai . Not sure why you didn't write it here, so I do :wink: .